src/tracing/errors.ts

```diff
@@ -0,0 +1,49 @@
+import * as Sentry from '@sentry/react'
+import { addExceptionMechanism } from '@sentry/utils'
+
+const ON_ERROR = 'onerror'
+const ON_UNHANDLED_REJECTION = 'onunhandledrejection'
+
+/** Adds a mechanism to events which have been sent from our global exception handlers. */
+export function beforeSendAddMechanism(event: Sentry.Event): Sentry.Event {
+  // Global exception handlers set an extra.mechanism, but the actual mechanism must be set on the event explicitly for
+  // Sentry to correctly process and display it in the Sentry UI. We still want mechanism to be set, despite using our
+  // own handler, because it aids in identifying unhandled errors and rejections.
+  if (event.extra?.mechanism) {
+    if (event.extra.mechanism === ON_ERROR || event.extra.mechanism === ON_UNHANDLED_REJECTION) {
+      addExceptionMechanism(event, { handled: false, type: event.extra.mechanism })
+      delete event.extra.mechanism // delete this so it doesn't clutter the Sentry UI
+    }
+  }
+
+  return event
+}
+
+export function onerror(event: Event | string, source?: string, lineno?: number, colno?: number, error?: Error) {
+  Sentry.captureException(error ?? event, { extra: { mechanism: ON_ERROR } })
+}
+
+export function onunhandledrejection({ reason }: { reason: unknown }) {
+  if (reason instanceof Error) {
+    if (isEthersRequestError(reason)) {
+      const method = JSON.parse(reason.requestBody).method
+
+      // ethers aggressively polls for block number, and it sometimes fails (whether spuriously or through rate-limiting).
+      // If it fails, it should not be considered an exception.
+      if (method === 'eth_blockNumber') return
+    }
+
+    // If the error is a network change, it should not be considered an exception, but should still be breadcrumbed.
+    if (reason.message.match(/underlying network changed/)) {
+      console.warn(reason) // logging adds the message to the breadcrumbs
```

Comment:
we're trying to avoid cluttering the console in prod with console logs. maybe use `debug` ?  warn is probably fine if you think it's important and doesn't happen too much though.

Reply:
I think this is a fair point tho @zzmp, what is the value of console logging in production? I always thought it was rather an anti-pattern. If we want to collect that data and act on it, we should probably collect it in a different way?